api/src/main/java/io/minio/CopyConditions.java

```diff
@@ -89,6 +89,20 @@ public void setMatchETagNone(String etag) throws InvalidArgumentException {
   }
 
   /**
+   * Set metadata directive to specify whether the metadata
+   * is copied from the source object or replaced with metadata
+   * provided in the request.
+   */
+  public void setMetadataDirective(String value) {
+    if (value == null) {
+      // http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectCOPY.html
+      // Default value is - COPY.
+      value = "COPY";
```

Comment:
As we provide a method to set a value for `x-amz-metadata-directive` header, its required to validate given arg.  String constants are not java style.  enum provides a strict typing.  By looking into javadoc, everybody knows what to pass to the method.